ports/stm/supervisor/port.c

```diff
@@ -237,16 +181,40 @@ safe_mode_t port_init(void) {
     _hrtc.Init.OutPut = RTC_OUTPUT_DISABLE;
 
     HAL_RTC_Init(&_hrtc);
-
     HAL_NVIC_EnableIRQ(RTC_Alarm_IRQn);
 
+    // Turn off SysTick
+    SysTick->CTRL = 0;
+
     return NO_SAFE_MODE;
 }
 
+void HAL_Delay(uint32_t delay_ms) {
+    if (SysTick->CTRL != 0) {
+        // SysTick is on, so use it
+        uint32_t tickstart = systick_ms;
+        while (systick_ms - tickstart < delay_ms) {
+        }
+    } else {
+        mp_hal_delay_ms(delay_ms);
+    }
+}
+
+uint32_t HAL_GetTick() {
+    if (SysTick->CTRL != 0) {
+        return systick_ms;
+    } else {
+        uint8_t subticks;
+        uint32_t result = (uint32_t)port_get_raw_ticks(&subticks);
```

Comment:
Do you want to account for the fact that a tick is 1/1024 of a second to CP? This is done because most LSEs are actually 32.768khz. Does the HAL expect a tick to be 1/1000th of a second?

Reply:
I saw your comment on discord and I don't think it matters too much. Every instance I've seen the HAL use these functions is to do this: 
```
uint32_t tickstart = HAL_GetTick();
while ((HAL_GetTick() - tickstart) <=  timout_dur) {
```
with a `timeout_dur` of 2 or 4 or something. It doesn't strike me as super precise, and the HAL even allows for the frequency and timeout durations to be tweaked. So I don't think the 2.3% difference is going to cause any differences in behavior.